test/reflect.js

```diff
@@ -0,0 +1,483 @@
+    });
+  });
+
+  describe('Reflect.ownKeys()', function () {
+    it('is a function', function () {
+      expect(typeof Reflect.ownKeys).to.equal('function');
+    });
+
+    it('throws if the target isn\'t an object', function () {
+      testPrimitiveThrow(function (item) {
+        return Reflect.ownKeys(item);
+      });
+    });
+
+    it('should return the same result as Object.getOwnPropertyNames if there are no Symbols', function () {
+      var obj = { foo: 1, bar: 2 };
+
+      obj[1] = 'first';
+
+      var result = Object.getOwnPropertyNames(obj);
+
+      expect(Reflect.ownKeys(obj)).to.deep.equal(result);
+      expect(result).to.deep.equal(['1', 'foo', 'bar']);
+    });
+
+    (typeof Symbol === 'function' ? it : xit)('symbols come last', function () {
+      var s = Symbol();
+
+      var o = {
+        'non-symbol': true
+      };
+
+      o[s] = true;
+
+      expect(Reflect.ownKeys(o)).to.deep.equal(['non-symbol', s]);
+    });
+  });
+
+  describe('Reflect.preventExtensions()', function () {
+    it('is a function', function () {
+      expect(typeof Reflect.preventExtensions).to.equal('function');
+    });
+
+    it('throws if the target isn\'t an object', function () {
+      testPrimitiveThrow(function (item) {
+        return Reflect.preventExtensions(item);
+      });
+    });
+
+    it('prevents extensions on objects', function () {
+      var obj = {};
+      Reflect.preventExtensions(obj);
+      expect(Object.isExtensible(obj)).to.equal(false);
+    });
+  });
+
+  describe('Reflect.set()', function () {
+    it('is a function', function () {
+      expect(typeof Reflect.set).to.equal('function');
+    });
+
+    it('throws if the target isn\'t an object', function () {
+      testPrimitiveThrow(function (item) {
+        return Reflect.set(item, 'prop', 'value');
+      });
+    });
+  });
+
+  describe('Reflect.setPrototypeOf()', function () {
+    it('is a function', function () {
+      expect(typeof Reflect.setPrototypeOf).to.equal('function');
+    });
+
+    it('throws if the target isn\'t an object', function () {
+      testPrimitiveThrow(function (item) {
+        return Reflect.setPrototypeOf(item, null);
+      });
+    });
+
+    it('throws if the prototype is neither object nor null', function () {
+      var o = {};
+
+      [undefined, 1, 'string', true].forEach(function (item) {
+        expect(function () {
+          return Reflect.setPrototypeOf(o, item);
+        }).to.throw(TypeError);
+      });
+    });
+
+    it('can set prototypes, and returns true on success', function () {
+      var obj = {};
+
+      expect(Reflect.setPrototypeOf(obj, Array.prototype)).to.equal(true);
+      expect(obj).to.be.an.instanceOf(Array);
+
+      expect(obj.toString).not.to.equal(undefined);
+      expect(Reflect.setPrototypeOf(obj, null)).to.equal(true);
+      expect(obj.toString).to.equal(undefined);
+    });
+
+    it('is returns false on failure', function () {
+      var obj = Object.freeze({});
```

Comment:
This test needs to be skipped in environments where `Object.freeze` is unavailable.